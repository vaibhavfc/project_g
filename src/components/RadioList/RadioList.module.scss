@import '../../sass/theme/default';

.root {
  color: $radio-circle;
  margin-bottom: -0.625rem;

  &:hover {
    background-color: rgba($radio-hover-unchecked, 0.08);
  }

  &:focus {
    background-color: rgba($radio-hover-unchecked, 0.08);
    transition: none;
  }
}

.checked {
  color: $radio-hover-checked;

  &:hover {
    background-color: rgba($radio-hover-checked, 0.08);
  }

  &:focus {
    background-color: rgba($radio-hover-checked, 0.08);
    transition: none;
  }
}

.disabledChecked {
  color: $radio-disabled;
}

.label {
  transform: translate(-0.375rem, 0.3125rem);
  color: $radio-label;
  @include body-2;

  // & > :first-of-type {
  //   padding: 0 0.5rem 0.125rem 0;
  // }
  // transform: translateY(-1rem);
}

.formLabel {
  color: $radio-label !important;
  text-transform: capitalize;
  @include body-3;
  // transform: translate(-0.5rem, -0.25rem);
}
