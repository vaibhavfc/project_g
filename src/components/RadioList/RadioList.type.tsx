import { Ref } from 'react'
import { RadioProps as MuiRadioProps, RadioClassKey } from '@material-ui/core/Radio'
export interface RadioListStylingProps extends Partial<Record<RadioClassKey, string>> {
}

export interface RadioListProps extends MuiRadioProps {
  // innerRef: Ref<HTMLButtonElement>;
  list: { title: string, 
          items: {
            label: string, 
            checked?: boolean, 
            disabled?: boolean
          }[]
        };
}
