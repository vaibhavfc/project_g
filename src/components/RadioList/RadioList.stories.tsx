// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import RadioList from '.';

import { RadioListProps } from './RadioList.type';

export default {
  title: 'Components/RadioList',
  component: RadioList,
  parameters: {
    controls: {
    },
  },
} as Meta;

const Template: Story<RadioListProps> = (args) => <RadioList {...args} />;

export const Default1 = Template.bind({});
Default1.args = {
  list: {
    title: 'Radio List Heading',
    items: [
      { label: 'radio-1', value: '1', name: 'radio' },
      { label: 'radio-2', value: '2', name: 'radio' },
      { label: 'radio-3', value: '3', name: 'radio' },
      { label: 'radio-4', value: '4', name: 'radio' },
    ],
  },
};

export const Default2 = Template.bind({});
Default2.args = {
  list: {
    title: 'Radio List Heading',
    defaultValue: '2',
    items: [
      { label: 'radio-1', value: '1', name: 'radio' },
      { label: 'radio-2', value: '2', name: 'radio' },
      { label: 'radio-3', value: '3', name: 'radio' },
      { label: 'radio-4', value: '4', name: 'radio' },
    ],
  },
};
