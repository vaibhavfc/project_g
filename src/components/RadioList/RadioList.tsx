
import React, { FC, forwardRef, useMemo } from 'react'
import { RadioListProps, RadioListStylingProps } from './RadioList.type'
import { Radio as MuiRadio, FormControl, FormControlLabel, FormGroup, FormHelperText, FormLabel, Typography } from "@material-ui/core"

import classes from './RadioList.module.scss'

import classnames from 'classnames';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const RadioList: FC<RadioListProps> = ({
  innerRef = null,
  classes: overrideClasses = {},
  children = 'RadioList',
  tabIndex= 0,
  list,
  ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<RadioListStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );

  // const [clicked, setClicked] = React.useState(false);

  // const handleChange = (event: React.ChangeEvent<HTMLButtonElement>) => {
  //   setClicked(event.target.checked);
  //   return clicked;
  // };

  return (
    <FormControl component="fieldset" className={classes.formControl}>
      <FormLabel component="legend"
        classes={{
          root: classnames(classes.formLabel),
          // focused: classnames(classes.formLabel)
        }}
      >
        {list.title}
      </FormLabel>
      <FormGroup>
        {
          list.items?.map(Radio => {
            return (
              <FormControlLabel
                classes={{
                  label: classes.label,
                }} 
                value="end"
                control={
                  <MuiRadio 
                    disabled={Radio.disabled} checked={Radio.checked} tabIndex={tabIndex}
                    classes={{
                      root: classes.root,
                      checked: classnames({
                        [classes.disabledChecked]: (Radio.disabled && Radio.checked),
                        [classes.checked]: (Radio.checked && !Radio.disabled),
                      })
                    }} 
                  />
                }
                label={Radio.label}
                labelPlacement="end"
              />  
            )
          })
        }
      </FormGroup>
    </FormControl> 
  )
}

// const RadioList = forwardRef<HTMLInputElement, Omit<RadioListProps, 'innerRef'>>(
//   (props, ref) => <RadioListWithInnerRef innerRef={ref} {...props} />
// );

export default RadioList;