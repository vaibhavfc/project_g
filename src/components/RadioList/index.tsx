import RadioList from './RadioList';
import { RadioListProps } from './RadioList.type';

export type { RadioListProps }

export default RadioList;