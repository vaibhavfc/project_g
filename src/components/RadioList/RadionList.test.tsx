import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import RadioList from './RadioList';

describe('test', () => {
  const list = {
    title: 'Checkbox List Heading',
    items: [
      { label: 'Default' },
      { label: 'Checked', checked: true },
      { label: 'Disabled', disabled: true },
      { label: 'Checked & Disabled', disabled: true, checked: true },
    ],
  };

  it('Renders RadioList Compoents', () => {
    render(<RadioList list={list}>test</RadioList>);
  });

  it('Matches Snapshot of RadioList', () => {
    const tree = renderer.create(<RadioList list={list}> TEST</RadioList>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
