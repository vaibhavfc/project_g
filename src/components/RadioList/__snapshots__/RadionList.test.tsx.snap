// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of RadioList 1`] = `
<fieldset
  className="MuiFormControl-root formControl"
>
  <legend
    className="MuiFormLabel-root formLabel"
  >
    Checkbox List Heading
  </legend>
  <div
    className="MuiFormGroup-root"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-11"
            disabled={false}
            onChange={[Function]}
            tabIndex={0}
            type="radio"
            value="end"
          />
          <div
            className="PrivateRadioButtonIcon-root-12"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label MuiTypography-body1"
      >
        Default
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-9 Mui-checked checked MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-11"
            disabled={false}
            onChange={[Function]}
            tabIndex={0}
            type="radio"
            value="end"
          />
          <div
            className="PrivateRadioButtonIcon-root-12 PrivateRadioButtonIcon-checked-14"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label MuiTypography-body1"
      >
        Checked
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root Mui-disabled"
    >
      <span
        aria-disabled={true}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-disabled-10 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-11"
            disabled={true}
            onChange={[Function]}
            tabIndex={0}
            type="radio"
            value="end"
          />
          <div
            className="PrivateRadioButtonIcon-root-12"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label Mui-disabled MuiTypography-body1"
      >
        Disabled
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root Mui-disabled"
    >
      <span
        aria-disabled={true}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-9 Mui-checked disabledChecked PrivateSwitchBase-disabled-10 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-11"
            disabled={true}
            onChange={[Function]}
            tabIndex={0}
            type="radio"
            value="end"
          />
          <div
            className="PrivateRadioButtonIcon-root-12 PrivateRadioButtonIcon-checked-14"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label Mui-disabled MuiTypography-body1"
      >
        Checked & Disabled
      </span>
    </label>
  </div>
</fieldset>
`;
